### DISCUSSION TITLE
Open Source Status of Mergify

### DISCUSSION THREAD

**Discussion by Xuanwo** [Mergifyio/mergify#5035] [Q&A]
Hi, I just noticed that mergify removed all the code from this repo in https://github.com/Mergifyio/mergify/pull/5031.

So mergify will be converted into a closed source software and not open source anymore? Are there any considerations behind this decision to share with us?

**Comment by jd** [comment]
Hi @Xuanwo 

There have been internal discussions about the pros and cons of keeping our engine open-source, and we moved future developments of our engine to a private repository a few weeks ago.

The main factor was no external contributor could contribute to the engine code as writing functional tests required a lot of undocumented setup with tons of special GitHub accounts. This makes it impracticable for anyone to contribute, which shows that there were only 2 contributions in 2021 from the same author, where Mergify's team had to write the tests themselves (so basically still doing 80% of the job).

Among other factors, we decided that building an open-source community around a code that nobody can contribute to and would neither run themselves is not worth it.

**Reply by Xuanwo** [reply]
Thanks for the explanation.

**Reply by ankush** [reply]
@jd 

I respect your decision but there's still argument for keeping FOSS projects FOSS even if they dont get many contributors - "Public Knowledge". I personally learned a thing or two from going through the code to learn how Mergify does backports of multi-commit PRs in a "smart way".

Thanks for keeping the history public though, perhaps archive it as separate repo so people can read it? Something similar to what reddit did: https://github.com/reddit-archive/reddit
